drivers/issi/is31fl3733.c

```diff
@@ -80,43 +80,58 @@ bool    g_pwm_buffer_update_required[DRIVER_COUNT] = {false};
 uint8_t g_led_control_registers[DRIVER_COUNT][24]             = {{0}, {0}};
 bool    g_led_control_registers_update_required[DRIVER_COUNT] = {false};
 
-void IS31FL3733_write_register(uint8_t addr, uint8_t reg, uint8_t data) {
+uint8_t IS31FL3733_write_register(uint8_t addr, uint8_t reg, uint8_t data) {
+    // If the transaction fails function returns 1.
     g_twi_transfer_buffer[0] = reg;
     g_twi_transfer_buffer[1] = data;
 
 #if ISSI_PERSISTENCE > 0
     for (uint8_t i = 0; i < ISSI_PERSISTENCE; i++) {
-        if (i2c_transmit(addr << 1, g_twi_transfer_buffer, 2, ISSI_TIMEOUT) == 0) break;
+        if (!(i2c_transmit(addr << 1, g_twi_transfer_buffer, 2, ISSI_TIMEOUT) == 0)) {
+            return 1;
+            break;
+        }
     }
 #else
-    i2c_transmit(addr << 1, g_twi_transfer_buffer, 2, ISSI_TIMEOUT);
+    if (i2c_transmit(addr << 1, g_twi_transfer_buffer, 2, ISSI_TIMEOUT)) {
+        return 1;
+    }
 #endif
+    return 0;
 }
 
-void IS31FL3733_write_pwm_buffer(uint8_t addr, uint8_t *pwm_buffer) {
-    // assumes PG1 is already selected
-
-    // transmit PWM registers in 12 transfers of 16 bytes
+uint8_t IS31FL3733_write_pwm_buffer(uint8_t addr, uint8_t *pwm_buffer) {
+    // Assumes PG1 is already selected.
+    // If any of the transactions fails function returns 1.
+    uint8_t output = 0;
+    // Transmit PWM registers in 12 transfers of 16 bytes.
     // g_twi_transfer_buffer[] is 20 bytes
 
-    // iterate over the pwm_buffer contents at 16 byte intervals
+    // Iterate over the pwm_buffer contents at 16 byte intervals.
     for (int i = 0; i < 192; i += 16) {
         g_twi_transfer_buffer[0] = i;
-        // copy the data from i to i+15
-        // device will auto-increment register for data after the first byte
-        // thus this sets registers 0x00-0x0F, 0x10-0x1F, etc. in one transfer
+        // Copy the data from i to i+15.
+        // Device will auto-increment register for data after the first byte
+        // Thus this sets registers 0x00-0x0F, 0x10-0x1F, etc. in one transfer.
         for (int j = 0; j < 16; j++) {
             g_twi_transfer_buffer[1 + j] = pwm_buffer[i + j];
         }
 
 #if ISSI_PERSISTENCE > 0
         for (uint8_t i = 0; i < ISSI_PERSISTENCE; i++) {
-            if (i2c_transmit(addr << 1, g_twi_transfer_buffer, 17, ISSI_TIMEOUT) == 0) break;
+            if (i2c_transmit(addr << 1, g_twi_transfer_buffer, 17, ISSI_TIMEOUT) == 0) {
+                break;
```

Comment:
I think this is the opposite of what you meant to do. You actually want to just set output = 1 and then break if it returns 0, Otherwise keep looping. no?

Reply:
ok so what we want here is:
If any i2c_transmit fails in IS31FL3733_write_pwm_buffer (i2c_transmit returns 1 is failure) we want IS31FL3733_write_pwm_buffer to return 1.